src/OrchardCore/OrchardCore.ResourceManagement/ResourceManager.cs

```diff
@@ -173,52 +171,14 @@ private ResourceDefinition FindResource(RequireSettings settings, bool resolveIn
             // using the action.
             var name = settings.Name ?? "";
             var type = settings.Type;
-            ResourceDefinition resource;
```

Comment:
I rewrote this without LINQ, code probably is even shorter now, this is potential bug source with setting version etc matching